sdk/go/common/apitype/plan.go

```diff
@@ -0,0 +1,89 @@
+package apitype
+
+import (
+	"encoding/json"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+)
+
+// PlanDiffV1 is the serializable version of a plan diff.
+type PlanDiffV1 struct {
+	// the resource properties that will be added.
+	Adds map[string]interface{} `json:"adds,omitempty"`
+	// the resource properties that will be deleted.
+	Deletes []string `json:"deletes,omitempty"`
+	// the resource properties that will be updated.
+	Updates map[string]interface{} `json:"updates,omitempty"`
+}
+
+// GoalV1 is the serializable version of a resource goal state.
+type GoalV1 struct {
+	// the type of resource.
+	Type tokens.Type `json:"type"`
+	// the name for the resource's URN.
+	Name tokens.QName `json:"name"`
+	// true if this resource is custom, managed by a plugin.
+	Custom bool `json:"custom"`
+	// the resource properties that will be changed.
+	CheckedInputs map[string]interface{} `json:"checkedInputs,omitempty"`
+	// the resource properties that will be changed.
+	InputDiff PlanDiffV1 `json:"inputDiff,omitempty"`
```

Comment:
Perhaps different doc comment between CheckedInputs vs InputDiff? A bit of elaboration here on how these are different could be really helpful to the reader. Presumably InputDiff=d(old, new), but are CheckedInputs total inputs in old or new version? Or they're partial?

Reply:
Ah yes, I should update CheckedInputs to point out its temporary as well until we get fixed autonames.